packages/core/admin/_internal/node/core/dependencies.ts

```diff
@@ -0,0 +1,199 @@
+ * and thus they are declared as peer dependencies – react / styled-components / etc.
+ *
+ * If these deps are not installed or declared, then we prompt the user to correct this. In
+ * V4 this is not a hard requirement, but in V5 it will be. Might as well get people started now.
+ */
+const checkRequiredDependencies = async ({
+  cwd,
+  logger,
+}: Pick<BuildOptions, 'cwd' | 'logger'>): Promise<CheckRequiredDependenciesResult> => {
+  const pkg = await readPkgUp({ cwd });
+
+  if (!pkg) {
+    throw new Error(`Could not find package.json at path: ${cwd}`);
+  }
+
+  logger.debug('Loaded package.json:', os.EOL, pkg.packageJson);
+
+  interface DepToReview {
+    name: string;
+    wantedVersion: string;
+    declaredVersion: string;
+  }
+
+  interface DepToInstall {
+    name: string;
+    wantedVersion: string;
+    declaredVersion?: never;
+  }
+
+  /**
+   * Run through each of the peer deps and figure out if they need to be
+   * installed or they need their version checked against.
+   */
+  const { install, review } = Object.entries(PEER_DEPS).reduce<{
+    install: DepToInstall[];
+    review: DepToReview[];
+  }>(
+    (acc, [name, version]) => {
+      if (!pkg.packageJson.dependencies) {
+        throw new Error(`Could not find dependencies in package.json at path: ${cwd}`);
+      }
+
+      const declaredVersion = pkg.packageJson.dependencies[name];
+
+      if (!declaredVersion) {
+        acc.install.push({
+          name,
+          wantedVersion: version,
+        });
+      } else {
+        acc.review.push({
+          name,
+          wantedVersion: version,
+          declaredVersion,
+        });
+      }
+
+      return acc;
+    },
+    {
+      install: [],
+      review: [],
+    }
+  );
+
+  if (install.length > 0) {
+    logger.info(
+      'The Strapi admin needs to install the following dependencies:',
+      os.EOL,
+      install.map(({ name, wantedVersion }) => `  - ${name}@${wantedVersion}`).join(os.EOL)
+    );
+
+    /**
+     * This prompt can be removed in V5 & therefore the code underneath can be refactored to
+     * only install the deps and return that we installed deps.
+     */
+    const { install: installAns } = await inquirer.prompt({
+      type: 'confirm',
+      name: 'install',
+      default: true,
+      message:
+        'Would you like to install these dependencies now? These are not required but are recommended, from V5 these will be required.',
+    });
+
+    if (installAns) {
+      /**
+       * TODO: Add an install deps command here. We need to resolve the package manager to do this & we also
+       * need to re-run this command because the node_modules shape could have changed post installation.
+       * So for now log an error and tell them to install them manually.
+       */
+      throw new Error(
+        'Please install the dependencies manually and then re-run this command, we have not implemented this yet.'
```

Comment:
It should also depend on the NODE_ENV as we shouldn't do this in prod or other non-dev env.

Reply:
I think you're right for for v4 because it's not required. I think in v5 these deps will be required so it's a different case - good spot!

Reply:
The main issue I see is prod deployment on read-only file-systems like GCP. We can't modify the filesystem at all so we shouldn't try to install anything. (Strapi Cloud doesn't strictly lock the filesystem)